### PR TITLE
Add stats tab

### PR DESCRIPTION
**Pull request by sebastiantoh**
image

@jialin7878 might have some merge conflicts with you cos I made some changes to the tabs (or rather, added the other tabs)

P.S. The tabs cannot be clicked unless the drawer is pulled open. Is this a desired behaviour?

TODO:
- integrate with backend

### CODE REVIEW

File: frontend/src/components/challengeProgressChart/ChallengeProgressChart.tsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+            y1="0"
+            x2="0"
+            y2="1"
+          >
+            <stop
+              offset="5%"
+              stopColor={color ?? '#8884d8'}
+              stopOpacity={0.8}
+            />
+            <stop offset="95%" stopColor={color ?? '#8884d8'} stopOpacity={0} />
```
Comment (ianyong) [inline frontend/src/components/challengeProgressChart/ChallengeProgressChart.tsx:83]: Can we extract this out to a variable?

File: frontend/src/pages/challenge/UserChallengeStats.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+const UserChallengeStats: React.FC<UserChallengeStatsProps> = (props) => {
+  const theme: Theme = useTheme();
+
+  const percentCompleted = 75;
+  const currentStreak = 4;
+  const longestStreak = 12;
+
+  const statsLabel = (
+    valueLabel: string,
+    desc: string,
```
Comment (ianyong) [inline frontend/src/pages/challenge/UserChallengeStats.tsx:47]: Can this be `description` instead? I read this as `descending` at first. 😅

File: frontend/src/pages/challenge/UserChallengeStats.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    return {
+      taskCompletionDate: add(mockFirstTaskDate, { days: i * 3 }),
+      taskIndex: i,
+    };
+  }),
+  totalNumberOfTasks: 10,
+  challengeEnrolledDate: mockChallengeEnrolledDate,
+};
+
+const UserChallengeStats: React.FC<UserChallengeStatsProps> = (props) => {
```
Comment (ianyong) [inline frontend/src/pages/challenge/UserChallengeStats.tsx:38]: This will be used eventually right (with regards to the warning)?

Comment (sebastiantoh) [reply frontend/src/pages/challenge/UserChallengeStats.tsx:38]: yep, it will be!

**Review by ianyong — APPROVED** [review]
LGTM!

**Pull request merged**